tsdb/engine/tsm1/reader.go

```diff
@@ -1350,7 +1354,7 @@ func (m *mmapAccessor) init() (*indirectIndex, error) {
 
 	indexOfsPos := len(m.b) - 8
 	indexStart := binary.BigEndian.Uint64(m.b[indexOfsPos : indexOfsPos+8])
-	if indexStart >= uint64(indexOfsPos) {
+	if indexStart >= uint64(indexOfsPos) || (uint64(indexOfsPos)-indexStart) > math.MaxInt32 {
```

Comment:
This added check ensures that index can be accessed using signed 32-bit offsets. Since `int32` index offsets are used internally (see `indirectIndex.UnmarshalBinary`), I think this is a correct check. I would prefer a distinct error message for this case.

Reply:
It does look like the correct check. The error message (all the error messages in this method) should include the file name, which we have through  `m.f.Name()`

Reply:
I think the most effective way to  wrap the file name onto error massage is the line 240 in tsdb/engine/tsm1/reader.go. 
What go you think @davidby-influx ?
```
index, err := t.accessor.init()
	if err != nil {
		**_return nil, err_**
	}
```